core-permission/src/main/java/com/thoughtworks/ark/core/permission/PermissionComposable.kt

```diff
@@ -0,0 +1,196 @@
+package com.thoughtworks.ark.core.permission
+
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberMultiplePermissionsState
+import com.google.accompanist.permissions.rememberPermissionState
+import com.thoughtworks.ark.ui.component.AppFilledButton
+import com.thoughtworks.ark.ui.theme.Theme
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class PermissionHandler {
+    internal val state = mutableStateOf(false)
+
+    fun requestPermission() {
+        state.value = true
+    }
+
+    internal fun resetState() {
+        state.value = false
+    }
+}
+
+class PermissionAlwaysDeniedHandler {
+    internal val state = mutableStateOf(false)
+    internal val alwaysDeniedState = mutableStateOf(false)
+
+    fun openSystemSettingPage() {
+        state.value = true
+    }
+
+    fun resetAlwaysDeniedState() {
+        alwaysDeniedState.value = false
+    }
+
+    internal fun resetState() {
+        state.value = false
+        alwaysDeniedState.value = false
+    }
+}
+
+@OptIn(ExperimentalPermissionsApi::class)
```

Comment:
I am not sure if the `Experimental` API is safe for our project.

As you know, the `Experiment` API may have some broken changes normally.

Reply:
As you know, many compose features are in the state of Experimental, but it can still be used normally, so this problem can be ignored